### PR TITLE
Correct typos and language

### PR DESCRIPTION
**Pull request by DannyvdSluijs**
This PR addresses several typos and language constructs found throughout the sources. I've groups the fixes in individual commits in an attempt to make review easiest. Making individual PR for each type of typo feels weird and one for each file feels like an overhead.

# Description
This PR address typos as found bij PHPStorm using the Problems view. Especially for the one where the introduction of a comma was suggested I've been withholding in addressing each issues as found by the editor. Especially for short sentences this would be not that helpfull.

## Suggested changelog entry
Correct typos and language

## Related issues/external references
_N/A_

## Types of changes

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
    - [ ] This change is only breaking for integrators, not for external standards or end-users.
- [X] Documentation improvement

## PR checklist

- [X] I have checked there is no other PR open for the same change.
- [X] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md).
- [X] I grant the project the right to include and distribute the code under the BSD-3-Clause license (and I have the right to grant these rights).
- [ ] I have added tests to cover my changes.
- [X] I have verified that the code complies with the projects coding standards.
- [ ] \[Required for new sniffs\] I have added XML documentation for the sniff.

### CODE REVIEW

**Comment by DannyvdSluijs** [comment]
I've noticed some test are failing which I'll address later tonight or at least this week.

**Comment by jrfnl** [comment]
@DannyvdSluijs Thanks for your willingness to contribute, but even just a quick glance at the changes, shows me quite a few are incorrect/introducing grammatical errors instead of fixing them.

Please do a very very very careful review of your proposed changes and update the branch to remove all the incorrect ones.

I'm tempted to close this PR as it is now, with an option to re-open it once you've had a chance to work out the kinks.

You may also want to consider splitting the PR in one which only touches docs files only, like the changelog and the readme and one which touches code files to make it more manageable.

**Comment by jrfnl** [comment]
@DannyvdSluijs Thank you for understanding. I'll leave some comments inline to point out things I noticed when I first had a quick look.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -302,7 +302,7 @@ the code should be flagged or not. Additionally, making auto-fixes to code in a
 
 This defensive coding in a sniff should, of course, also be safeguarded via tests.
 
-Parse/compile errors can take various forms. Some can be handled without much problems by PHPCS, some can not.
+Parse/compile errors can take various forms. Some can be handled without many problems by PHPCS, some can not.
```
Comment (jrfnl) [inline .github/CONTRIBUTING.md:305]: "without much problems" is an expression and is correct. "without many problems" changes the meaning of the sentence.

File: reproductions/3875.php [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<?php
```
Comment (jrfnl) [inline reproductions/3875.php:1]: This file doesn't belong here.

File: src/Standards/MySource/Sniffs/Channels/IncludeSystemSniff.php [hunk 3]
```diff
@@ -114,7 +114,7 @@ protected function processTokenWithinScope(File $phpcsFile, $stackPtr, $currScop
             $name = $this->getIncludedClassFromToken($phpcsFile, $tokens, $i);
             if ($name !== false) {
                 $includedClasses[$name] = true;
-                // Special case for Widgets cause they are, well, special.
+                // Special case for Widgets because they are, well, special.
```
Comment (jrfnl) [inline src/Standards/MySource/Sniffs/Channels/IncludeSystemSniff.php:117]: I'd accept this change, but `cause` is valid English, though maybe colloquial.

File: src/Standards/PEAR/Sniffs/WhiteSpace/ScopeClosingBraceSniff.php [hunk 4]
```diff
@@ -59,7 +59,7 @@ public function process(File $phpcsFile, $stackPtr)
         $scopeEnd   = $tokens[$stackPtr]['scope_closer'];
 
         // If the scope closer doesn't think it belongs to this scope opener
-        // then the opener is sharing its closer with other tokens. We only
+        // then the opener is sharing it's closer with other tokens. We only
```
Comment (jrfnl) [inline src/Standards/PEAR/Sniffs/WhiteSpace/ScopeClosingBraceSniff.php:62]: `its` is correct.

File: src/Standards/PEAR/Tests/Commenting/FileCommentUnitTest.1.inc [hunk 5]
```diff
@@ -17,7 +17,7 @@ declare(encoding='utf-8');
 * that is available through the world-wide-web at the following URI:
 * http://www.php.net/license/3_0.txt.  If you did not receive a copy of
 * the PHP License and are unable to obtain it through the web, please
-* send a note to license@php.net so we can mail you a copy immediately.
+* send a note to license@php.net, so we can mail you a copy immediately.
```
Comment (jrfnl) [inline src/Standards/PEAR/Tests/Commenting/FileCommentUnitTest.1.inc:20]: This is a predefined text in the tests and should not be changed.

File: src/Tokenizers/Tokenizer.php [hunk 6]
```diff
@@ -1355,7 +1355,7 @@ private function recurseScopeMap($stackPtr, $depth=1, &$ignore=0)
                         || ($openerNested === true
                         && $this->tokens[$i]['nested_parenthesis'] !== $this->tokens[$stackPtr]['nested_parenthesis'])
                     ) {
-                        // We found the a token that looks like the opener, but it's nested differently.
+                        // We found the token that looks like the opener, but it's nested differently.
```
Comment (jrfnl) [inline src/Tokenizers/Tokenizer.php:1358]: The "the" should be removed, the "a" is correct.

File: src/Tokenizers/Tokenizer.php [hunk 7]
```diff
@@ -1554,7 +1554,7 @@ private function createLevelMap()
                     // conditions where there was none. This happens for T_CASE
                     // statements that are using the same break statement.
                     if ($lastOpener !== null && $this->tokens[$lastOpener]['scope_closer'] === $this->tokens[$i]['scope_closer']) {
-                        // This opener shares its closer with the previous opener,
+                        // This opener shares it's closer with the previous opener,
```
Comment (jrfnl) [inline src/Tokenizers/Tokenizer.php:1557]: `its` is correct.

File: src/Util/Standards.php [hunk 8]
```diff
@@ -238,14 +238,14 @@ public static function isInstalledStandard($standard)
...
-            // standards directory.
+            // standards' directory.
             $standard = Common::realPath($standard);
             if ($standard === false) {
                 return false;
             }
 
             // Might be an actual ruleset file itUtil.
-            // If it has an XML extension, let's at least try it.
+            // If it has an XML extension, lets at least try it.
```
Comment (jrfnl) [inline src/Util/Standards.php:248]: I believe the `let's` is correct here, but wouldn't mind a second opinion.

File: tests/ConfigDouble.php [hunk 9]
```diff
@@ -8,7 +8,7 @@
  *
  * This class is a "double" of the Config class which prevents this from happening.
  * In _most_ cases, tests should be using this class instead of the "normal" Config,
- * with the exception of select tests for the Config class itself.
+ * except select tests for the Config class itself.
```
Comment (jrfnl) [inline tests/ConfigDouble.php:11]: Nothing wrong with the original phrasing.

File: tests/Core/File/GetMethodParametersTest.php [hunk 10]
```diff
@@ -1982,7 +1982,7 @@ public function testPHP8ConstructorPropertyPromotionWithTypes()
 
 
     /**
-     * Verify recognition of PHP8 constructor with both property promotion as well as normal parameters.
+     * Verify recognition of PHP8 constructor with both property promotion and normal parameters.
```
Comment (jrfnl) [inline tests/Core/File/GetMethodParametersTest.php:1985]: This changes the meaning (though subtly).

File: tests/Core/File/GetTokensAsStringTest.php [hunk 11]
```diff
@@ -102,7 +102,7 @@ public function testLengthBeyondEndOfFile()
      * Test getting a token set as a string.
      *
      * @param string     $testMarker     The comment which prefaces the target token in the test file.
-     * @param int|string $startTokenType The type of token(s) to look for for the start of the string.
+     * @param int|string $startTokenType The type of token(s) to look for the start of the string.
```
Comment (jrfnl) [inline tests/Core/File/GetTokensAsStringTest.php:105]: This change is incorrect. The original may look strange with the double `for`, but is correct. (same below)

**Comment by jrfnl** [comment]
@DannyvdSluijs I haven't gone through all of it, but just left some pointers now. Hope they help.

**Pull request closed without merging**